src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -96,6 +96,8 @@ internal DefaultVisualStudioRazorParser(RazorCodeDocument codeDocument)
 
         public override ITextBuffer TextBuffer => _documentTracker.TextBuffer;
 
+        public override bool IsIdle => _pendingStructureChanges == 0 && _parser.IsIdle;
```

Comment:
Can the state of this property change on a background thread? If that's possible then this isn't super useful

Reply:
It can because it's based off of the background parser that's how it's always been. This is something we can definitely change by enforcing changes to be enqueued on the foreground thread and then tracking the "latest" change until its completion.

Reply:
This just seems like it's a big buggy by definition if that's the case.

Running a work item on the foreground thread is a bit like taking a lock. You expect that the things you are interacting with won't change while you're *holding the lock*